build/generate-style-code.js

```diff
@@ -39,14 +35,16 @@ global.flowType = function (property) {
 };
 
 global.propertyType = function (property) {
-    if (isDataDriven(property)) {
-        return `DataDrivenProperty<${flowType(property)}>`;
-    } else if (/-pattern$/.test(property.name) || property.name === 'line-dasharray') {
-        return `CrossFadedProperty<${flowType(property)}>`;
-    } else if (property.name === 'heatmap-color' || property.name === 'line-gradient') {
-        return `ColorRampProperty`;
-    } else {
-        return `DataConstantProperty<${flowType(property)}>`;
+    switch (property['property-type']) {
+        case 'data-driven':
+            return `DataDrivenProperty<${flowType(property)}>`;
+        case 'cross-faded':
+            return `CrossFadedProperty<${flowType(property)}>`;
+        case 'color-ramp':
+            return `ColorRampProperty`;
+        case 'data-constant':
+        default:
+            return `DataConstantProperty<${flowType(property)}>`;
```

Comment:
Just for completeness, should we also list `case 'constant':` here?  (Alternatively, since `DataConstantProperty<T>` isn't really intended to represent a totally constant property, it could be `case 'constant': throw new Error("Constant properties unimplemented");`)